### PR TITLE
Bugfix DIS-854 [v118] Post-migration pocket errors caused by Firefox iOS clients sending invalid locales

### PR DESCRIPTION
**Pull request by nbhasin2**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

## :bulb: Description
In this PR we updated locales to match the latest valid list of pocket locales and added a check to ensure only to call the backend if both the feature and locale are enabled.

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by lmarceau — APPROVED** [review]

**Comment by nbhasin2** [comment]
cc @dnarcese keeping you in the loop so you know which locale we are shipping, I have asked folks from the Pocket team to add it to the ticket for reference so marking it `Do not merge` until those are finalized. 

I want to say they are but better to have them in the ticket as well.

File: Providers/Pocket/PocketProvider.swift [hunk 1]
```diff
@@ -30,7 +30,14 @@ class PocketProvider: PocketStoriesProviding, FeatureFlaggable, URLCaching {
...
     private let pocketEnvAPIKey = "PocketEnvironmentAPIKey"
 
-    private static let SupportedLocales = ["en_CA", "en_US", "en_GB", "en_ZA", "de_DE", "de_AT", "de_CH"]
+    private static let SupportedLocales = [
+        "fr_FR",
+        "es_ES",
+        "it_IT",
+        "en_CA", "en_GB", "en_US",
+        "de_DE", "de_AT", "de_CH"
+    ]
```
Comment (mmiermans) [inline Providers/Pocket/PocketProvider.swift:39]: Let's keep this unchanged for now, to not block this bug-fix by a discussion about expanding to additional markets.
```suggestion
    private static let SupportedLocales = ["en_CA", "en_US", "en_GB", "en_ZA", "de_DE", "de_AT", "de_CH"]
```

**Review by mmiermans — COMMENTED** [review]
Thanks for the quick response! I believe the `guard` in `fetchStories` will fix the issue.

File: Providers/Pocket/PocketProvider.swift (hunk 1 above)
Comment (nbhasin2) [reply Providers/Pocket/PocketProvider.swift]: Perfect, reverted it back to what it was before, we can revisit it if we want to support more locales

**Pull request merged**